### DISCUSSION TITLE
BGW on managed databases.

### DISCUSSION THREAD

**Discussion by jorenvandeweyer** [pgpartman/pg_partman#364] [General]
My database is hosted and managed by a cloud provider which provides the `pg_partman` extension.

When I'm not executing the `partman.run_maintenance` function myself manually. It looks like the `SuperUser` of the cloud provider is executing this function in my database.

I'm making these assumptions because when I list the child tables of my partition, the `SuperUser` user of the cloud provider is the owner of these tables.

Now I'm wondering if they are executing the function "manually" or if they have the background worker enabled, and if so, with which settings. How can I verify this?

**Comment by jorenvandeweyer** [comment]
This might be preventable by setting `inherit_privileges` to `true` (https://github.com/pgpartman/pg_partman/issues/355#issuecomment-822530899)

**Reply by keithf4** [reply]
Inherit privileges isn't related to what your original question is asking nor the BGW. That just sets whether the child tables inherit the privileges (GRANTS,REVOKES, Ownership) that exists on the parent table. By default, native partitioning does not inherit those things.

**Reply by jorenvandeweyer** [reply]
I guess my comment was pretty vague.

Some context:
I executed  `partman.run_maintenance` from my own database user. Which then threw an error:

```
Query 1 ERROR: ERROR:  permission denied for table raw_p2021_05_26
CONTEXT: SQL statement "SELECT max("createdAt")::text FROM public.raw_p2021_05_26"
PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 213 at EXECUTE
DETAIL: 
HINT: 
CONTEXT:  PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 415 at RAISE
```

I'm thinking this might be preventable by the `inherit_privileges` setting.

**Comment by keithf4** [comment]
If they are using the BGW there should be settings in the postgresql.conf to indicate that that is what's being used. Otherwise, you'd have to ask the cloud provider how they are doing that.

**Reply by jorenvandeweyer** [reply]
I guess I'll have to open a support ticket then. I don't have any permissions to view to config file.

```
ERROR:  must be superuser or a member of pg_read_all_settings to examine "config_file"
```
